portfolio/src/main/webapp/script.js

```diff
@@ -92,3 +91,200 @@ function deleteComments() {
+        'Known hisotric temples include the Imperial Palace and the Meiji Shinto Shrine, which is known for its towering gate and surrounding woods. ' + 
+        'Tokyo also features two distinctive towers: Tokyo Tower, and the new Tokyo Skytree, which is the tallest tower in the world.</p>';
+
+    const tkInfo = new google.maps.InfoWindow({content: tkStr});
+
+    tkMarker.addListener('click', function() {
+        tkInfo.open(map, tkMarker);
+    });
+
+    /** San Francisco */
+    const sfLatlng = {lat: 37.7749, lng: -122.4194};
+
+    const sfMarker = new google.maps.Marker({
+        position: sfLatlng,
+        map: map,
+        title: "San Francisco"
+    });
+
+    const sfStr = '<p><b>San Francisco</b> is the cultural, commercial, and financial center of Northern California. ' + 
+        'It is known for its cool summers, fog, steep rolling hills, eclectic mix of architecture, and landmarks, including the Golden Gate Bridge, Fisherman\'s Wharf, ' + 
+        'and the former Alcatraz Federal Penitentiary. It is also the headquarters of five major banking institutions and various other companies.</p>';
+
+    const sfInfo = new google.maps.InfoWindow({content: sfStr});
+
+    sfMarker.addListener('click', function() {
+        sfInfo.open(map, sfMarker);
+    });
+
+    /** Sapporo */
+    const spLatlng = {lat: 43.0618, lng: 141.3545};
+
+    const spMarker = new google.maps.Marker({
+        position: spLatlng,
+        map: map,
+        title: "Sapporo"
+    });
+
+    const spStr = '<p><b>Sapporo</b>, capital of the mountainous northern Japanese island of Hokkaido, is famous for its beer, ' + 
+        'skiing and annual Sapporo Snow Festival featuring enormous ice sculptures. It is the fifth largest city in Japan. ' + 
+        'Ski hills and jumps from the 1972 Winter Olympics are scattered within the city limits, and the Sapporo Beer Museum traces the city’s brewing history.</p>';
+
+    const spInfo = new google.maps.InfoWindow({content: spStr});
+
+    spMarker.addListener('click', function() {
+        spInfo.open(map, spMarker);
+    });
+
+    /** Hangzhou */
+    const hzLatlng = {lat: 30.2741, lng: 120.1551};
+
+    const hzMarker = new google.maps.Marker({
+        position: hzLatlng,
+        map: map,
+        title: "Hangzhou"
+    });
+    
+    const hzStr = '<p><b>Hangzhou</b>, the capital of China’s Zhejiang province, is the southern terminus of the ancient ' + 
+        'Grand Canal waterway, which originates in Beijing. It is known for its West Lake, which is celebrated by poets and artists since the 9th century ' + 
+        'and encompasses islands, temples, pavilions, gardens and arched bridges.</p>';
+
+    const hzInfo = new google.maps.InfoWindow({content: hzStr});
+
+    hzMarker.addListener('click', function() {
+        hzInfo.open(map, hzMarker);
+    });
+}
```

Comment:
it would be better to create a json array/object or json file that stores all of this information. Then you can loop through the object and create InfoWindows and add listeners for each location.

Reply:
fixed it on the Blobstore branch (which will be merged to this branch:))